### ISSUE TITLE
Error compiling magithub after deleting the elpa folder

### ISSUE THREAD

**Issue by shrysr** [jkitchin/scimax#206]
I just deleted my elpa folder and re-started scimax. However, there appears to be some issue with magithub and it does not compile successfully. Therefore, I am unable to load the basic scimax configuration itself. The entire compile log is attached and the portion at the end refers to magithub.  I'd also be glad to know work-arounds to these situations. 

[compile-error-emacs.txt](https://github.com/jkitchin/scimax/files/2041110/compile-error-emacs.txt)

**Comment by jkitchin** [comment]
Maybe just comment out the magithub line in packages.el. That will probably at least let you get back to work.

**Comment by shrysr** [comment]
Thanks. I figured this out from the other issues, but then surprisingly met another error.  
`require: Symbol’s value as variable is void: ivy-height-alist`

**Comment by jkitchin** [comment]
that should be defined in counsel.

**Comment by shrysr** [comment]
I did not make modifications to any other packages, and would not have expected any further errors. However, after several repeated attempts of deleting the elpa folder and restarting scimax - the issue was resolved today, with magithub enabled. I guess the issue can be closed, unless you have any further comments. 

Note: I'm not sure why (or if it helped) - but there were several emacs-daemon processes when I looked at `ps aux | grep emacs`, and I killed them.

**Comment by jkitchin** [comment]
Sometimes a package is corrupt in Melpa, and when melpa is rebuilt (every 4ish hours) it gets fixed. 

Lately I have not been able to use emacsclient reliably. It seems to hang, and lead to multiple processes like you describe. I always start emacs this way now:

emacs -q -l /path/to/scimax/init.el

and I don't have those problems. I am not sure what changes led to this issue.

I use 

pkill -9 Emacs 

to kill them. They always caused problems for me, creating lock files, etc. In any case, glad to hear things are running again.

**Comment by shrysr** [comment]
Thank you. I do appreciate the insight. Yes, I've been feeling some lag in the start up of the emacsclient as of late. However, since I've been playing around with the config, and have a 8 year old ancient laptop - it's hard to judge and it hasn't felt crippling yet. 

The main reason I stick to the daemon is that I try to use different frames for different workflows / projects on virtual desktops in the mac, and I've always loved the instant load of the emacsclient. Even so, I have seen that using 'too many' frames does make emacs-plus crash.

Do you use something like workgroups with frame /  window configurations? This seems to be the correct way forward, but I have not got it working the way I would like. Would love advice on this.

I used to include the elpa folder with scimax, with the idea of 'increasing reliability' and maintaining atleast one 'absolutely reliable' backup config, no matter what happened. However, I guess my definition of stable is always evolving with Emacs. Should I revert to the above method?

I've become so used to scimax now that vanilla emacs for just a temporary basis is still irksome. Yesterday, I actually tried Atom due to the above 'crisis', and discarded it.

**Comment by jkitchin** [comment]
I haven't felt too much pain without the client. I tend to always have emacs open anyway, and rarely use the terminal. I use projectile, and it is easy to switch between projects with that,  including opening files in new frames.  I have tried a variety of workgroup/window config things, none of which have stuck. You can save window configs in registers pretty easy, in scimax with the hydra menu it is: f12 rwn, where n is the key to save the configuration in. Then later, you type f12 rjn to jump to that register. These are not persistent. I think you can also save these with ivy (https://oremacs.com/2016/06/27/ivy-push-view/). For a while, I used the hotspots command, but I am not using it as much. there remain many options...

**Comment by shrysr** [comment]
I tried ivy-push-view, but it appears complex to retain the window configuration after a restart, and I had trouble with getting what I desired. The Scimax hydra menu is a nice option, and I do use hotspots frequently. 
However, I've managed to get workgroups going to some extent. I use a session with saved wconfigs within workgroups. I use the snippet below along with some modeline additions. The session does not  automatically load when I start a emacs daemon and then a new frame. Once I load the session manually - things work reasonably as expected. There's an exit-save behavior, but I'm not yet sure if that works as expected.

`
(require 'workgroups2)
(setq wg-session-load-on-start t)       ; default: (not (daemonp))

;; Change prefix key (before activating WG)
(setq wg-prefix-key (kbd "C-c z"))

;; Change workgroups session file
(setq wg-session-file "~/my_gits/scimax-personal/user/workgroups/main-session-wg")

(setq wg-emacs-exit-save-behavior           'save)      ; Options: 'save 'ask nil
(setq wg-workgroups-mode-exit-save-behavior 'save)      ; Options: 'save 'ask nil
(workgroups-mode 1)   ; put this one at the bottom of .emacs
`
